### ISSUE TITLE
Assignments to stack variables are sometimes dropped

### ISSUE THREAD

**Issue by simonlindholm** [matt-kempster/mips_to_c#31]
See e.g. tests/irix-g/output/multiple-assigns.c, which reads from sp4, but that variable is never declared or written to.

**Comment by simonlindholm** [comment]
08f7df42f1294ba9495fea1968f3a695b3326262 should solve most of these cases, but there's one case it doesn't handle. Quoting from the code:
```
# TODO: This is wrong when arguments are
# stored to the stack, onto other variables than their default stack
# spilling locations... In that case it hides assignments.
```

**Comment by simonlindholm** [comment]
Fixed in 4e7c9cec774e36de814ad01fda2adff2e70f43c9.
